File: Employee/validation.py
from pickle import TRUE
from cerberus import Validator

def validation_insert(body):
   v = Validator()
   v.schema = {
         'id': { 
            'type': 'integer',
            'minlength': 1
                     },
         'first_name': { 
            'type': 'string',
            'minlength': 2
                     },
         'last_name': { 
            'type': 'string',
            'minlength': 2
                     },
         "email": {
            "type": "string",
            "minlength": 8,
            "maxlength": 255,
            "regex": "^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+$"
                  },
         'gender': {
            'type': 'string',
                   },
         "phone": {
            "type": "string",
            "minlength": 10,
            "maxlength": 11,
            "regex": "^0[0-9]{9}$"
                  },
         "password": {
            "type": "string",
            "minlength": 8,
            "maxlength": 50
                  }
            }
   
   val= v.validate(body)
   if val:
      return True
   else:
      return v.errors


def validation_update(body):
   v = Validator()
   v.schema = {
         'id': { 
            'type': 'integer',
            'minlength': 1
                     },
         'first_name': { 
            'type': 'string',
            'minlength': 2
                     }
                  
            } 
   val= v.validate(body)
   if val:
      return True
   else:
      return v.errors


def validation_insert_id(body):
   v = Validator()
   v.schema = {
         'id': { 
            'type': 'string',
            'minlength': 1,
            "regex": "[0-9]+"
                     }                  
            } 
   val= v.validate(body)
   if val:
      return True
   else:
      return v.errors


def validation_delete(body):
   v = Validator()
   v.schema = {
         'id': { 
            'type': 'string',
            'minlength': 1,
            "regex": "[0-9]+"
                     }                  
            } 
   val= v.validate(body)
   if val:
      return True
   else:
      return v.errors

def validation_check(body):
   v = Validator()
   v.schema = {
         "email": {
            "type": "string",
            "minlength": 8,
            "maxlength": 255,
            "regex": "^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+$"
                  },
         "password": {
            "type": "string",
            "minlength": 8,
            "maxlength": 50
                  }
   }
   val= v.validate(body)
   if val:
      return True
   else:
      return v.errors

File: Employee/validation.js
const joi = require('joi')

const inputUser = joi.object({
  id: joi.number().min(1).max(100),
  first_name: joi.string().min(1).max(10),
  last_name: joi.string().min(1).max(10),
  email: joi.string().email().lowercase(),
  gender: joi.string().valid("m", "f", "o"),
  phone: joi.number().min(1000000000).max(9999999999),
  password: joi.string().pattern(new RegExp("^[a-zA-Z0-9]{3,30}$")).min(4).max(15),
}).unknown(true);

const signinUser = joi.object({
  email: joi.string().email().lowercase(),
  password: joi.string().pattern(new RegExp("^[a-zA-Z0-9]{3,30}$")).min(4).max(15),
});

const adduservalidation = async (req, res, next) => {
  const value = await inputUser.validate(req.body);
  if (value.error) {
    res.json({
      success: 0,
      message: value.error.details[0].message
    })
  } else {
    next();
  }
};

const addsigninvalidation = async (req, res, next) => {
  const value = await signinUser.validate(req.body);
  if (value.error) {
    res.json({
      success: 0,
      message: value.error.details[0].message
    })
  } else {
    next();
  }
};


const arrayOfSchema = joi.array().items(inputUser)


const validateArrayofObjects = async (req, res, next) => {
  const value = await arrayOfSchema.validate(req.body,{abortEarly: false});
  if (value.error) {
    // for (let i = 0; i < value.error.details.length ; i++) 
    //   {
      console.log("$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$")
      console.log(value.error._original.value)
        res.json({
          message: value.error.message
          //message: value.error._original
      })
      // res.json({
      //   message: value.error._original
      // })
    // }
  } else {
    next();
  }
};





module.exports = {
  adduservalidation,
  addsigninvalidation,
  validateArrayofObjects
}